src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -318,7 +319,7 @@ export class EvaluationPlugin extends UIPlugin {
     const cell = this.getters.getCell(position);
 
     if (cell && cell.content) {
-      if (cell.isFormula && !isBadExpression(cell.content)) {
+      if (cell.isFormula && !isBadExpression(cell.compiledFormula.tokens)) {
```

Comment:
maybe we can recombine/reorder those two ifs and check `cell.isFormula` first. The goal would be to avoid `cell.content` for formulas since it isn't completely free since it rebuilds the range strings.